mdata/reorder_buffer_test.go

```diff
@@ -0,0 +1,333 @@
+package mdata
+
+import (
+	"fmt"
+	"reflect"
+	"testing"
+
+	"gopkg.in/raintank/schema.v1"
+)
+
+func testAddAndGet(t *testing.T, reorderWindow uint32, testData, expectedData []schema.Point, expectAddFail bool) []schema.Point {
+	var flushed []schema.Point
+	b := NewReorderBuffer(reorderWindow, 1)
+	metricsTooOld.SetUint32(0)
+	for _, point := range testData {
+		addRes := b.Add(point.Ts, point.Val)
+		flushed = append(flushed, addRes...)
+	}
+	if expectAddFail && metricsTooOld.Peek() == 0 {
+		t.Fatal("Expected an add to fail, but they all succeeded")
+	}
+	returned := b.Get()
+
+	if !reflect.DeepEqual(expectedData, returned) {
+		t.Fatal(fmt.Sprintf("Returned data does not match expected data\n%+v\n %+v", testData, expectedData))
```

Comment:
That seems to make the tests fail https://circleci.com/gh/raintank/metrictank/2413